### ISSUE TITLE
application cannot be started

### ISSUE THREAD

**Issue by kmorber** [Quedale/OnvifDeviceManager#17]
kmorber@sparky:/usr/local/bin/OnvifDeviceManager$ ./onvifmgr 

(onvifmgr:148839): dbind-WARNING **: 21:29:22.677: Couldn't connect to accessibility bus: Failed to connect to socket /root/.cache/at-spi/bus_0: Keine Berechtigung
DEBUG 24-02-04 21:29:22.678318 140473623648960 onvifinitstaticplugins.c:223: Initializing Gstreamer plugins...
DEBUG 24-02-04 21:29:22.678442 140473623648960 onvifinitstaticplugins.c:228: Loading static plugins...
DEBUG 24-02-04 21:29:22.684406 140473623648960 onvifinitstaticplugins.c:435: Gstreamer plugins initialized...
WARN  24-02-04 21:29:22.684413 140473623648960 onvif-mgr.c:72: Element 'avdec_h264' not found.
WARN  24-02-04 21:29:22.684417 140473623648960 onvif-mgr.c:72: Element 'avdec_h265' not found.
WARN  24-02-04 21:29:22.684423 140473623648960 onvif-mgr.c:80: Found element 'openh264dec'. Increasing priority from 64 to 65...
DEBUG 24-02-04 21:29:22.684428 140473623648960 onvif-mgr.c:49: *** video/x-h264 decoders ***
DEBUG 24-02-04 21:29:22.684482 140473623648960 onvif-mgr.c:57: *    openh264dec[65]
DEBUG 24-02-04 21:29:22.684490 140473623648960 onvif-mgr.c:60: ****************************
DEBUG 24-02-04 21:29:22.684494 140473623648960 onvif-mgr.c:49: *** video/x-h265 decoders ***
DEBUG 24-02-04 21:29:22.684529 140473623648960 onvif-mgr.c:57: *    libde265dec[128]
DEBUG 24-02-04 21:29:22.684534 140473623648960 onvif-mgr.c:60: ****************************
DEBUG 24-02-04 21:29:22.684540 140473623648960 onvif-mgr.c:49: *** image/jpeg decoders ***
DEBUG 24-02-04 21:29:22.684570 140473623648960 onvif-mgr.c:57: *    jpegdec[256]
DEBUG 24-02-04 21:29:22.684576 140473623648960 onvif-mgr.c:60: ****************************
DEBUG 24-02-04 21:29:22.684580 140473623648960 onvif-mgr.c:49: *** video/x-av1 decoders ***
DEBUG 24-02-04 21:29:22.684610 140473623648960 onvif-mgr.c:60: ****************************
INFO  24-02-04 21:29:22.684615 140473623648960 onvif-mgr.c:113: Using Gstreamer Version : 1.22.9.0
TRACE 24-02-04 21:29:22.684621 140473623648960 app_dialog.c:65: init
TRACE 24-02-04 21:29:22.708127 140473623648960 credentials_input.c:28: create
TRACE 24-02-04 21:29:22.708142 140473623648960 app_dialog.c:65: init
TRACE 24-02-04 21:29:22.708596 140473623648960 msg_dialog.c:60: create
TRACE 24-02-04 21:29:22.708603 140473623648960 app_dialog.c:65: init
TRACE 24-02-04 21:29:22.708945 140473623648960 event_queue.c:83: create...
INFO  24-02-04 21:29:22.711476 140473623648960 app_settings.c:188: Reading Settings file onvifmgr_settings.ini
WARN  24-02-04 21:29:22.711488 140473623648960 app_settings.c:245: No config file found. Using default configs. 1
DEBUG 24-02-04 21:29:22.712469 140473623648960 src_retriever.c:11: Test Audio Src [pulsesrc] device [(null)]
DEBUG 24-02-04 21:29:22.713054 140473623648960 alsa_devices.c:11: Create AlsaDevices list
DEBUG 24-02-04 21:29:22.713098 140473623648960 alsa_utils.c:21: **** List of Hardware Devices CAPTURE****
DEBUG 24-02-04 21:29:22.713845 140473623648960 alsa_utils.c:49: card 0: UACDemoV10 [UACDemoV1.0], device 0: USB Audio [USB Audio]
DEBUG 24-02-04 21:29:22.713853 140473623648960 alsa_devices.c:74: Create AslaDevice object
DEBUG 24-02-04 21:29:22.713922 140473623648960 alsa_utils.c:49: card 2: Generic_1 [HD-Audio Generic], device 0: ALC897 Analog [ALC897 Analog]
DEBUG 24-02-04 21:29:22.713932 140473623648960 alsa_devices.c:74: Create AslaDevice object
DEBUG 24-02-04 21:29:22.713938 140473623648960 alsa_utils.c:49: card 2: Generic_1 [HD-Audio Generic], device 2: ALC897 Alt Analog [ALC897 Alt Analog]
DEBUG 24-02-04 21:29:22.713941 140473623648960 alsa_devices.c:74: Create AslaDevice object
DEBUG 24-02-04 21:29:22.714122 140473623648960 src_retriever.c:11: Test Audio Src [alsasrc] device [hw:0,0]
free(): invalid next size (fast)
Abgebrochen

**Comment by Quedale** [comment]
I've reviewed and updated alsa related source files.
Can you pull the latest change and give that a shot?

If it still crashes, please run it using valgrind to extract additional details.
`valgrind ./onvifmgr`

Thanks for your time reporting this.
